driver/linux/main.c

```diff
@@ -299,6 +301,19 @@ static int __init sgx_init(void)
 
 	if (!sgx_page_cache_init())
 		return -EFAULT;
+#ifdef HAVE_MMPUT_ASYNC
+	k_mmput_async = mmput_async;
+#else
+#ifdef HAVE_KSYM_LOOKUP
+	k_mmput_async = (void*)kallsyms_lookup_name("mmput_async");
+#else
+	#error "kernel version is not supported. We need either mmput_async or kallsyms_lookup_name exported from the kernel"
```

Comment:
I suppose we could also do this check in the Makefile, although that wouldn't help simplifying this code.

Reply:
right